core/src/test/java/gov/nist/secauto/metaschema/core/metapath/function/library/FnStringJoinTest.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * SPDX-FileCopyrightText: none
+ * SPDX-License-Identifier: CC0-1.0
+ */
+
+package gov.nist.secauto.metaschema.core.metapath.function.library;
+
+import static gov.nist.secauto.metaschema.core.metapath.TestUtils.string;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import gov.nist.secauto.metaschema.core.metapath.ExpressionTestBase;
+import gov.nist.secauto.metaschema.core.metapath.IMetapathExpression;
+import gov.nist.secauto.metaschema.core.metapath.item.ISequence;
+import gov.nist.secauto.metaschema.core.metapath.item.atomic.IStringItem;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+class FnStringJoinTest
+    extends ExpressionTestBase {
+  private static Stream<Arguments> provideValues() { // NOPMD - false positive
+    return Stream.of(
+        Arguments.of(
+            string("123456789"),
+            "string-join(1 to 9)"),
+        Arguments.of(
+            string("Now is the time ..."),
+            "string-join(('Now', 'is', 'the', 'time', '...'), ' ')"),
+        Arguments.of(
+            string(""),
+            "string-join((), 'separator')"),
+        Arguments.of(
+            string("1, 2, 3, 4, 5"),
+            "string-join(1 to 5, ', ')")
+    // Arguments.of(
+    // string("xml:id=\"xyz\""),
+    // "let $doc := <doc><chap><section xml:id=\"xyz\"/></chap></doc>\n"
+    // + "return $doc//@xml:id ! fn:string-join((node-name(), '=\"', ., '\"'))"
+    // ),
+    // Arguments.of(
+    // string("doc/chap/section"),
+    // "let $doc := <doc><chap><section xml:id=\"xyz\"/></chap></doc>\n"
+    // + "return $doc//@xml:id ! fn:string-join((node-name(), '=\"', ., '\"'))"
+    // )
+    );
```

Comment:
@david-waltermire, I just wanted to check but even using `metachema-cli metapath eval -e "let $doc := <doc><chap><section xml:id=\"xyz\"/></chap></doc>"` with a current develop build throws weird errors from compilation and the output messages even swallowing the var name whether it be `$doc` or `$ex` or anything else. Because this data is not model-bound is that even supported or easily testable in the function like this, or should be tested in an alternative approach?

```
Unable to compile Metapath 'let  := <doc><chap><section xml:id="xyz"/></chap></doc>'
org.antlr.v4.runtime.misc.ParseCancellationException: line 1:5 mismatched input ':=' expecting <EOF>
        at gov.nist.secauto.metaschema.core.metapath.antlr.FailingErrorListener.syntaxError(FailingErrorListener.java:25) ~[dev.metaschema.java.metaschema-core-2.2.0-SNAPSHOT.jar:?]
        at org.antlr.v4.runtime.ProxyErrorListener.syntaxError(ProxyErrorListener.java:41) ~[org.antlr.antlr4-runtime-4.13.2.jar:4.13.2]
        at org.antlr.v4.runtime.Parser.notifyErrorListeners(Parser.java:544) ~[org.antlr.antlr4-runtime-4.13.2.jar:4.13.2]
        at org.antlr.v4.runtime.DefaultErrorStrategy.reportInputMismatch(DefaultErrorStrategy.java:327) ~[org.antlr.antlr4-runtime-4.13.2.jar:4.13.2]
        at org.antlr.v4.runtime.DefaultErrorStrategy.reportError(DefaultErrorStrategy.java:139) ~[org.antlr.antlr4-runtime-4.13.2.jar:4.13.2]
        at gov.nist.secauto.metaschema.core.metapath.antlr.Metapath10.metapath(Metapath10.java:221) ~[dev.metaschema.java.metaschema-core-2.2.0-SNAPSHOT.jar:?]
        at gov.nist.secauto.metaschema.core.metapath.MetapathExpression.compile(MetapathExpression.java:80) ~[dev.metaschema.java.metaschema-core-2.2.0-SNAPSHOT.jar:?]
        at gov.nist.secauto.metaschema.core.metapath.IMetapathExpression.compile(IMetapathExpression.java:136) ~[dev.metaschema.java.metaschema-core-2.2.0-SNAPSHOT.jar:?]
        at gov.nist.secauto.metaschema.cli.commands.metapath.EvaluateMetapathCommand.executeCommand(EvaluateMetapathCommand.java:195) ~[dev.metaschema.java.metaschema-cli-2.2.0-SNAPSHOT.jar:?]
        at gov.nist.secauto.metaschema.cli.processor.command.ICommandExecutor.lambda$using$0(ICommandExecutor.java:46) ~[dev.metaschema.java.cli-processor-2.2.0-SNAPSHOT.jar:?]
        at gov.nist.secauto.metaschema.cli.processor.CLIProcessor$CallingContext.invokeCommand(CLIProcessor.java:521) [dev.metaschema.java.cli-processor-2.2.0-SNAPSHOT.jar:?]
        at gov.nist.secauto.metaschema.cli.processor.CLIProcessor$CallingContext.processCommand(CLIProcessor.java:497) [dev.metaschema.java.cli-processor-2.2.0-SNAPSHOT.jar:?]
        at gov.nist.secauto.metaschema.cli.processor.CLIProcessor.parseCommand(CLIProcessor.java:234) [dev.metaschema.java.cli-processor-2.2.0-SNAPSHOT.jar:?]
        at gov.nist.secauto.metaschema.cli.processor.CLIProcessor.process(CLIProcessor.java:210) [dev.metaschema.java.cli-processor-2.2.0-SNAPSHOT.jar:?]
        at gov.nist.secauto.oscal.tools.cli.core.CLI.runCli(CLI.java:83) [dev.metaschema.oscal.oscal-cli-enhanced-2.5.0-SNAPSHOT.jar:?]
        at gov.nist.secauto.oscal.tools.cli.core.CLI.main(CLI.java:48) [dev.metaschema.oscal.oscal-cli-enhanced-2.5.0-SNAPSHOT.jar:?]
gov.nist.secauto.metaschema.core.metapath.StaticMetapathException: MPST0003: Unable to compile Metapath 'let  := <doc><chap><section xml:id="xyz"/></chap></doc>'
alexanderjstein@FCOH2J-JH2LXQ2H fedramp-automation % ~/opt/bin/oscal-cli metaschema metapath eval -e "let $ex := <doc><chap><section xml:id=\"xyz\"/></chap></doc>" 
Unable to compile Metapath 'let  := <doc><chap><section xml:id="xyz"/></chap></doc>'
org.antlr.v4.runtime.misc.ParseCancellationException: line 1:5 mismatched input ':=' expecting <EOF>
        at gov.nist.secauto.metaschema.core.metapath.antlr.FailingErrorListener.syntaxError(FailingErrorListener.java:25) ~[dev.metaschema.java.metaschema-core-2.2.0-SNAPSHOT.jar:?]
        at org.antlr.v4.runtime.ProxyErrorListener.syntaxError(ProxyErrorListener.java:41) ~[org.antlr.antlr4-runtime-4.13.2.jar:4.13.2]
        at org.antlr.v4.runtime.Parser.notifyErrorListeners(Parser.java:544) ~[org.antlr.antlr4-runtime-4.13.2.jar:4.13.2]
        at org.antlr.v4.runtime.DefaultErrorStrategy.reportInputMismatch(DefaultErrorStrategy.java:327) ~[org.antlr.antlr4-runtime-4.13.2.jar:4.13.2]
        at org.antlr.v4.runtime.DefaultErrorStrategy.reportError(DefaultErrorStrategy.java:139) ~[org.antlr.antlr4-runtime-4.13.2.jar:4.13.2]
        at gov.nist.secauto.metaschema.core.metapath.antlr.Metapath10.metapath(Metapath10.java:221) ~[dev.metaschema.java.metaschema-core-2.2.0-SNAPSHOT.jar:?]
        at gov.nist.secauto.metaschema.core.metapath.MetapathExpression.compile(MetapathExpression.java:80) ~[dev.metaschema.java.metaschema-core-2.2.0-SNAPSHOT.jar:?]
        at gov.nist.secauto.metaschema.core.metapath.IMetapathExpression.compile(IMetapathExpression.java:136) ~[dev.metaschema.java.metaschema-core-2.2.0-SNAPSHOT.jar:?]
        at gov.nist.secauto.metaschema.cli.commands.metapath.EvaluateMetapathCommand.executeCommand(EvaluateMetapathCommand.java:195) ~[dev.metaschema.java.metaschema-cli-2.2.0-SNAPSHOT.jar:?]
        at gov.nist.secauto.metaschema.cli.processor.command.ICommandExecutor.lambda$using$0(ICommandExecutor.java:46) ~[dev.metaschema.java.cli-processor-2.2.0-SNAPSHOT.jar:?]
        at gov.nist.secauto.metaschema.cli.processor.CLIProcessor$CallingContext.invokeCommand(CLIProcessor.java:521) [dev.metaschema.java.cli-processor-2.2.0-SNAPSHOT.jar:?]
        at gov.nist.secauto.metaschema.cli.processor.CLIProcessor$CallingContext.processCommand(CLIProcessor.java:497) [dev.metaschema.java.cli-processor-2.2.0-SNAPSHOT.jar:?]
        at gov.nist.secauto.metaschema.cli.processor.CLIProcessor.parseCommand(CLIProcessor.java:234) [dev.metaschema.java.cli-processor-2.2.0-SNAPSHOT.jar:?]
        at gov.nist.secauto.metaschema.cli.processor.CLIProcessor.process(CLIProcessor.java:210) [dev.metaschema.java.cli-processor-2.2.0-SNAPSHOT.jar:?]
        at gov.nist.secauto.oscal.tools.cli.core.CLI.runCli(CLI.java:83) [dev.metaschema.oscal.oscal-cli-enhanced-2.5.0-SNAPSHOT.jar:?]
        at gov.nist.secauto.oscal.tools.cli.core.CLI.main(CLI.java:48) [dev.metaschema.oscal.oscal-cli-enhanced-2.5.0-SNAPSHOT.jar:?]
gov.nist.secauto.metaschema.core.metapath.StaticMetapathException: MPST0003: Unable to compile Metapath 'let  := <doc><chap><section xml:id="xyz"/></chap></doc>'
```

Reply:
Inline XML in Metapath statements is not supported by the library. There is no way to parse this without an associated Metaschema module.

Reply:
OK, just checking. Is there a desire to make "Metachema instance equivalents" for this kind of test? I assume not, just checking.